docs/how-to-guides/monitoring-with-prometheus-operator.md

```diff
@@ -152,6 +159,85 @@ kubectl -n monitoring get secret prometheus-operator-grafana -o jsonpath='{.data
 
 Open the following URL: `https://grafana.<cluster name>.<DNS zone>`. Enter the username `admin` and the password obtained from the previous step.
 
+## Add custom Grafana dashboards
+
+Create a ConfigMap with keys as the dashboard file names and values as JSON dashboard. See the following example:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: grafana-dashboards
+  namespace: myapp
+  labels:
+    grafana_dashboard: "true"
+data:
+  grafana-dashboard1.json: |
+    {
+      "annotations": {
+[REDACTED]
+```
+
+Add the label `grafana_dashboard: "true"` so that grafana automatically picks up the dashboards in the ConfigMaps across the cluster.
+
+This can also be done by using the following two imperative commands:
+
+```bash
+kubectl -n myapp create cm grafana-dashboards \
+  --from-file=grafana-dashboard1.json \
+  --from-file=grafana-dashboard2.json \
+  --dry-run -o yaml | kubectl apply -f -
+
+kubectl -n myapp label cm grafana-dashboards grafana_dashboard=true
+```
+
+## Add new ServiceMonitors
+
+### Default Prometheus operator setting:
+
+Create a ServiceMonitor with the required configuration and make sure to add the following label, so that the prometheus-operator will track it:
+
+```yaml
+metadata:
+  labels:
+    release: prometheus-operator
+```
+
+### Custom Prometheus operator setting:
+
+Deploy the prometheus-operator with the following setting, and it watches all ServiceMonitors across the cluster:
+
+```tf
+watch_labeled_service_monitors = "false"
+```
+
+Then there is no need to add any label to ServiceMonitor, at all. Create a ServiceMonitor, and prometheus-operator tracks it.
+
+## Add custom alerts for Alertmanager
+
+### Default Prometheus operator setting:
+
+Create a PrometheuRule object with the required configuration and make sure to add the following labels, so that prometheus-operator will track it:
+
+```yaml
+metadata:
+  labels:
+    release: prometheus-operator
+    app: prometheus-operator
+```
+
+### Custom Prometheus operator setting:
+
+Deploy the prometheus-operator with the following setting, and it watches all PrometheusRules across the cluster:
+
+```tf
+watch_labeled_prometheus_rules = "false"
+```
+
+Then there is no need to add any label to PrometheusRule, at all. Create a PrometheusRule, and prometheus-operator tracks it.
+
```

Comment:
I don't fancy having  this paragraph at the very bottom of the document.

I mean if the user wants the above setting then it essentially means reinstalling the Prometheus operator with the above change included, which makes the flow of the document bottom to top, instead of top to bottom.

It would be nice if the block from  `Add new ServiceMonitors` until the end are added before we provide the instruction to the user `lokoctl component apply prometheus-operator`.

Reply:
From my POV, these are specialised settings that users get to only when they have gone through basic things like quickly getting the prometheus operator up and running.

Reply:
agreed, but mentioned settings make following all the above steps feel unimportant.

If its before the component is installed, then the user has a choice whether to go for such a setting or not.
If its at the last, the component is already installed and others steps are already done.

Reply:
Installation of prometheus operator or any other component is idempotent so even if we make changes later and apply again the user will see the effects of it by doing apply again.

What I don't want this doc is to feel like a steep learning curve by throwing all the things they might not want.

These sections are at the end because these are the use cases people come to on day 2 and not on day 1. For instance when someone is learning on how to do monitoring with Lokomotive they will do steps like installing with most basic settings, then trying to get hold of all the sub components, then they will think of how to monitor their own apps with the existing monitoring setup we have.

IMHO showing custom settings to user even before installing is not a good idea.

Reply:
noted. 

Might we add something like `For custom settings on AlertManager and ServiceMonitors check this link` , something similar ?

Reply:
Do you mean, like we put a link to sub-section up in the doc?

Reply:
yes.

Reply:
done.